spacy/scorer.py

```diff
@@ -295,20 +295,23 @@ def score_spans(
             # Find all predidate labels, for all and per type
             gold_spans = set()
             pred_spans = set()
-            # Special case for ents:
-            # If we have missing values in the gold, we can't easily tell
-            # whether our NER predictions are true.
-            # It seems bad but it's what we've always done.
-            if attr == "ents" and not all(token.ent_iob != 0 for token in gold_doc):
-                continue
             for span in getter(gold_doc, attr):
                 gold_span = (span.label_, span.start, span.end - 1)
                 gold_spans.add(gold_span)
                 gold_per_type[span.label_].add((span.label_, span.start, span.end - 1))
             pred_per_type = {label: set() for label in labels}
-            for span in example.get_aligned_spans_x2y(getter(pred_doc, attr)):
-                pred_spans.add((span.label_, span.start, span.end - 1))
-                pred_per_type[span.label_].add((span.label_, span.start, span.end - 1))
+            align_x2y = example.alignment.x2y
+            for pred_span in getter(pred_doc, attr):
+                indices = align_x2y[pred_span.start : pred_span.end].dataXd.ravel()
+                if len(indices):
+                    g_span = gold_doc[indices[0] : indices[-1]]
+                    # Check we aren't missing annotation on this span. If so,
+                    # our prediction is neither right nor wrong, we just
+                    # ignore it.
+                    if any(token.ent_iob == 0 for token in g_span):
```

Comment:
Is this backwards?

Reply:
It's what I intended to write (although maybe my thinking is wrong!)

My idea is that we fetch the gold span, and if we don't know any of its annotations, we should skip this `pred_span` and not evaluate it.

This means we'll always by necessity be checking the accuracy of all the gold spans. If we don't recover any of them, we'll lose accuracy for that. But we don't lose precision for predicting spans that map onto tokens that have missing annotations.

Reply:
Oh wait, yes it is! Thanks.